libgalois/include/katana/PropertyGraph.h

```diff
@@ -262,38 +262,51 @@ class KATANA_EXPORT PropertyGraph {
     rdg_.set_mirror_nodes(std::move(a));
   }
 
+  /// Return the node property table for local nodes
+  const std::shared_ptr<arrow::Table>& node_properties() const {
+    return rdg_.node_properties();
+  }
+
+  /// Return the edge property table for local edges
+  const std::shared_ptr<arrow::Table>& edge_properties() const {
+    return rdg_.edge_properties();
+  }
+
 public:
   /// PropertyView provides a uniform interface when you don't need to
   /// distinguish operating on edge or node properties
   struct PropertyView {
     PropertyGraph* g;
+    const PropertyGraph* const_g;
 
     std::shared_ptr<arrow::Schema> (PropertyGraph::*schema_fn)() const;
-    std::shared_ptr<arrow::ChunkedArray> (PropertyGraph::*property_fn)(
+    std::shared_ptr<arrow::ChunkedArray> (PropertyGraph::*property_fn_int)(
         int i) const;
-    const std::shared_ptr<arrow::Table>& (
-        PropertyGraph::*properties_fn)() const;
+    std::shared_ptr<arrow::ChunkedArray> (PropertyGraph::*property_fn_str)(
+        const std::string& str) const;
+    int32_t (PropertyGraph::*property_num_fn)() const;
     Result<void> (PropertyGraph::*add_properties_fn)(
         const std::shared_ptr<arrow::Table>& props);
     Result<void> (PropertyGraph::*upsert_properties_fn)(
         const std::shared_ptr<arrow::Table>& props);
     Result<void> (PropertyGraph::*remove_property_int)(int i);
     Result<void> (PropertyGraph::*remove_property_str)(const std::string& str);
 
-    std::shared_ptr<arrow::Schema> schema() const { return (g->*schema_fn)(); }
-
-    std::shared_ptr<arrow::ChunkedArray> Property(int i) const {
-      return (g->*property_fn)(i);
+    std::shared_ptr<arrow::Schema> schema() const {
+      return (const_g->*schema_fn)();
     }
 
-    const std::shared_ptr<arrow::Table>& properties() const {
-      return (g->*properties_fn)();
+    std::shared_ptr<arrow::ChunkedArray> GetProperty(int i) const {
+      return (const_g->*property_fn_int)(i);
     }
 
-    std::vector<std::string> property_names() const {
-      return properties()->ColumnNames();
+    std::shared_ptr<arrow::ChunkedArray> GetProperty(
+        const std::string& str) const {
+      return (const_g->*property_fn_str)(str);
     }
 
+    int32_t GetPropertyNum() const { return (const_g->*property_num_fn)(); }
```

Comment:
idle comment: every time I read this, I always think "shouldn't this be called GetNumProperties", but I think the last time this came up, we decided the current `GetFooNum` was consistent with the current API so we have kept things as there are now (cc @roshandathathri who is the person I strongly associate with this discussion)

Reply:
I have updated the names to `GetNumProperties()`.